conans/client/conan_api.py

```diff
@@ -39,12 +39,20 @@
 from conans.util.tracer import log_command, log_exception
 
 
+def get_basic_requester(client_cache):
+    requester = requests.Session()
+    requester.proxies = client_cache.conan_config.proxies
+    return requester
+
 def api_method(f):
     def wrapper(*args, **kwargs):
         the_self = args[0]
         try:
             log_command(f.__name__, kwargs)
             with tools.environment_append(the_self._client_cache.conan_config.env_vars):
+                # Patch the globals in tools
+                tools._global_requester = get_basic_requester(the_self._client_cache)
```

Comment:
What? you need to patch the module, so users importing tools in their conanfiles got the patched version. I do it in the ConanAPI because is the lowest endpoint right now.